src/librustc_mir/borrow_check/mod.rs

```diff
@@ -473,11 +473,20 @@ impl<'cx, 'gcx, 'tcx> DataflowResultsConsumer<'cx, 'tcx> for MirBorrowckCtxt<'cx
                 );
             }
             StatementKind::ReadForMatch(ref place) => {
-                self.access_place(
+                // Read for match doesn't access any memory and is used to
```

Comment:
Which borrows was this conflicting with? It feels like we *should* be able to treat this as an access. Maybe the `ReadForMatch` statements are coming at the wrong place?

Oh, I guess I see, this is not really about the shallow borrows we are making, but rather about the "read for match" that we add to the "main path" being matched?

(Which may or may not have a shallow borrow associated with it)

Reply:
This is for the discriminant => ReadForMatch change (although I might change this when ReadForMatch is split)